src/csharp/Grpc.Tools.MSBuildIntegration/Grpc.Tools.MSBuildIntegration.targets

```diff
@@ -0,0 +1,27 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <CoreCompileDependsOn>Proto;$(CoreCompileDependsOn)</CoreCompileDependsOn>
+  </PropertyGroup>
+  <ItemGroup>
+    <AvailableItemName Include="Proto"/>
+  </ItemGroup>
+  <Target Name="Proto" Inputs="@(Proto)" Outputs="@(Proto->'$(IntermediateOutputPath)%(FileName).cs')">
+    <!-- Run our platform independent wrapper that launches protoc with the right arguments -->
+    <Exec Condition="'$(OS)' == 'Windows_NT'" Command="$(MSBuildThisFileDirectory)..\tools\Grpc.Tools.CSharpGenerator.exe $(ProjectDir) $(IntermediateOutputPath) @(Proto->'%(filename)%(extension)', ' ')" LogStandardErrorAsError="true"/>

```

Comment:
@jtattermusch: This is only possible if we either
a) Use tracker. Tracker is an amazingly underdocumented tool, and exists only in Microsoft package of MSBuild. Since it is not in github/microsoft/MSBuild, it is not likely to exist in mono. I have code that fully supports tracker (I wrote it to hande complext dependencies in a 3rd part compiler) and can release it, but it will not be cross-platform
b) Create a managed, .NET version of protoc parser that will read files and tell us what files it is going to generate. Sounds more complex but portable. But you and protoc team will have to maintain these in sync. Nasty.
c) Sorry, no (c) that I can think of at the moment. Let's think if there is a better solution.